python/pyarrow/_parquet.pyx

```diff
@@ -52,6 +52,9 @@ _MAX_ROW_GROUP_SIZE = 64*1024*1024
 cdef class Statistics(_Weakrefable):
     """Statistics for a single column in a single row group."""
 
+    def __init__(self):
+        raise TypeError(f"Can't instantiate internal class {self.__class__.__name__}")
```

Comment:
Those classes are not really internal, though, it's just that they shouldn't be instantiated directly. Can we change the error message to something else? For example:
```suggestion
        raise TypeError(f"Do not call {self.__class__.__name__}'s constructor directly")
```